raiden/transfer/secret_registry.py

```diff
@@ -14,16 +20,34 @@ def events_for_onchain_secretreveal(
     secret: Secret,
     expiration: BlockExpiration,
     block_hash: BlockHash,
+    target_state: TargetTransferState = None,
+    pseudo_random_generator: random.Random = None
```

Comment:
can you please explain the reason why we need the RNG only now and not before?

Reply:
It's needed to generate the message_id

Reply:
this was discussed in person and it is way outside of the scope for this PR.